csharp/src/Drivers/Databricks/Lz4Utilities.cs

```diff
@@ -77,13 +87,13 @@ public static ReadOnlyMemory<byte> DecompressLz4(byte[] compressedData, int buff
 
         /// <summary>
         /// Asynchronously decompresses LZ4 compressed data into memory.
-        /// Returns the buffer and length as a tuple for efficient wrapping in a MemoryStream.
+        /// Returns a RecyclableMemoryStream that must be disposed by the caller.
         /// </summary>
         /// <param name="compressedData">The compressed data bytes.</param>
         /// <param name="cancellationToken">Cancellation token for the async operation.</param>
-        /// <returns>A tuple containing the decompressed buffer and its valid length.</returns>
+        /// <returns>A MemoryStream containing the decompressed data. Caller must dispose.</returns>
         /// <exception cref="AdbcException">Thrown when decompression fails.</exception>
-        public static Task<(byte[] buffer, int length)> DecompressLz4Async(
+        public static Task<MemoryStream> DecompressLz4Async(
```

Comment:
If these are always returning a `RecyclableMemoryStream` instead of a `MemoryStream`, perhaps the method should be typed to do that? One important distinction is that a lot of people don't bother disposing a `MemoryStream` because it's a NOP, but failure to `Dispose` a `RecyclableMemoryStream` means that the blocks of memory aren't returned to the manager until the finalizer runs -- which might be considerably later.